lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -1608,6 +1678,17 @@ instance FromJSON ApiByronWalletBalance where
 instance ToJSON ApiByronWalletBalance where
     toJSON = genericToJSON defaultRecordTypeOptions
 
+instance FromJSON ApiWalletAssetsBalance where
+    parseJSON = genericParseJSON defaultRecordTypeOptions
+instance ToJSON ApiWalletAssetsBalance where
+    toJSON = genericToJSON defaultRecordTypeOptions
+
+-- fixme: doesn't quite match the spec
+instance FromJSON (ApiT W.TokenMap) where
+    parseJSON = fmap (ApiT . W.getFlat) . parseJSON
+instance ToJSON (ApiT W.TokenMap) where
+    toJSON = toJSON . W.Flat . getApiT
```

Comment:
Hi @rvl 

I'm just curious about the reason for choosing `Flat` here. Is there a reason to not use `Nested`?

(From recollection, the Daedalus team indicated that they preferred a nested representation, where assets were grouped by token policy.)

Reply:
Nesting makes it more complicated to process. The fact that there's a hierarchy policy_id -> asset_name is actually an implementation detail. End-users care about tokens, and tokens are uniquely identified by a pair (policy_id, asset_name). Tokens with different names are different which is why in the end, a flat views of assets makes makes more sense in the interface.

Reply:
Best place to comment on the API would be #2446. But it's one line of javascript to convert between flat and nested. EIther way would probably be fine.